File: 2019/solutions/prob05.java
import java.util.Scanner;

public class prob05 {

	public static void main(String[] args) {
		int lowerCase = 0;
		int upperCase = 0;
		Scanner input = new Scanner(System.in);
		String s = input.nextLine();
		s = s.replaceAll("[^a-zA-Z]", "");
		
		for(int i = 0; i < s.length(); i++)
			if(Character.isUpperCase(s.charAt(i)))
				upperCase++;
			else
				lowerCase++;
		if(upperCase == lowerCase)
			System.out.println("true");
		else
			System.out.println("false");
		input.close();
	}

}

File: 2019/solutions/prob05.js
const readline = require('readline');

const readInterface = readline.createInterface({
    input: process.stdin,
    console: false
});

readInterface.on('line', function(line) {
    let lower = 0;
    let upper = 0;

    // Count number of lowercase and uppercase letters
    for(var x = 0; x < line.length; x++){
        let char = line.charAt(x);
        if ('a' <= char && char <= 'z') lower++;
        if ('A' <= char && char <= 'Z') upper++;
    }

    //Determine if balanced or not
    if (lower == upper) {
        console.log('true');
    } else {
        console.log('false');
    }

    //Repeat for each input line
});
